### ISSUE TITLE
TypeError: can't convert CUDA tensor to numpy. Use Tensor.cpu() to copy the tensor to host memory first.

### ISSUE THREAD

**Issue by JeremyKeusters** [Hzzone/pytorch-openpose#8]
First of all: thanks for your amazing work! I would like to use OpenPose, but can't run Caffe in my current setup/environment. This repo is a life-saver.

Now to the point: I got the following error when running the demo.py:
`TypeError: can't convert CUDA tensor to numpy. Use Tensor.cpu() to copy the tensor to host memory first.`

I fixed this by changing the line:
`output = self.model(data).numpy()`

to:
`output = self.model(data).cpu().numpy()`

As numpy can't work on stuff that is still on the GPU.. I'm not sure if this needs to be changed in the repo and if this is the best solution for this error? Thanks.

My environment specs are the following:
OS: CentOS Linux release 7.5.1804
CUDA version: 8.0
cuDNN version: 7.0.5
Torch version: 0.4.1
Python version: Python 3.6.6
GPU's: 4x GeForce RTX 2070

**Comment by Hzzone** [comment]
```
output = self.model(data).cpu().numpy()
```

**Comment by JeremyKeusters** [comment]
Yup, that’s the same solution as I posted above. So you agree that this is the right solution?

**Comment by 12345k** [comment]
@JeremyKeusters  your solution has helped me. Thanks

**Comment by swithmn1** [comment]
output = self.model(data).cpu().numpy()；this is right!
